### ISSUE TITLE
[BUG] After ZeroQuant compression, the model size didn't decrease.

### ISSUE THREAD

**Issue by dianyo** [microsoft/DeepSpeed#2608]
Hi,

I wanted to do ZeroQuant on my SwinTransformer, to see if ZeroQuant works for it. I copied the ds_config in `DeepSpeedExamples/model_compression/bert/config/ZeroQuant/ds_config_W8A8_Qgroup48_fp32.json` and only modified the quantized modules to my model's modules. However, after I ran `init_compression` and `redundancy_clean`, the model size was almost unchanged when I saved the final model. The following is my code:
```
if args.compression:
    assert args.path_to_model is not None
    model.load_state_dict(torch.load(args.path_to_model))
    model = init_compression(model, args.deepspeed_config)
	
	model = redundancy_clean(model, args.deepspeed_config)
    print ('after_clean')
    model_to_save = model.module if hasattr(model, 'module') else model
    torch.save(model_to_save.state_dict(), args.saving_folder + '/clean_net.pkl')
```

Does anyone have any idea what went wrong? Or did I get something wrong for the ZeroQuant?

**Comment by yaozhewei** [comment]
Hi, 

redundancy cleaning won't save the model into a compact version but help user to check the simulated quantization accuracy. If you a compressed version, you will need to save the weight into INT8 + its Scaling Factor. 

Best,

**Comment by dianyo** [comment]
Hi @yaozhewei ,

Thanks for your reply. Now, I understand that I have to save the model by converting the weight into INT8 format after `redundancy_clean ` function. However, I have no idea how I can get the scaling factor for each group. Is there any attribute that can get the values?

**Comment by yaozhewei** [comment]
Yes, it might be easier if you do not apply "redundancy clean" but directly convert the model's parameter from FP16/32 to INT8 by looping the model.named_parameters() (select the ones you want to convert), and use a modification of this function https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/compression/utils.py#L88 to convert the parameter to INT8. You will need to modify the line I highlighted to extract the scaling factor.

**Comment by yaozhewei** [comment]
Close the issue as there is no followup. Please reopen it if you need further assistance

**Comment by iwkkk** [comment]
I have the same problem and I still don't understand how to save the int8 model.  Is there anyone know how to do with that?

**Comment by aakejiang** [comment]
So is there any tool that can produce a INT8 quantized model using ZeroQuant?

**Comment by MatteoPagliani** [comment]
Hi @yaozhewei, is there a way to directly save the model with int8 weights and quantization statistics (scale factors and zero-points)?
